### PR TITLE
Fixes PropType issues #124 and #125

### PR DESCRIPTION
**Pull request by ctrlplusb**
Please don't be afraid to tell me off for doing anything stupid or in an inefficient manner.  Your skills eclipse mine! :)

### CODE REVIEW

**Comment by ctrlplusb** [comment]
Looks like subject references don't work, so:

Fixes #124 and #125

**Comment by ctrlplusb** [comment]
Hmmm, just realised the way I am doing default props extension might be wrong...

File: src/index.js [hunk 1]
```diff
@@ -129,11 +129,40 @@ for (let i=ELEMENTS.length; i--; ) {
 }
 
 
+function isStatelessComponent(c) {
```
Comment (developit) [inline src/index.js]: Thinking we should make this match [preact's existing implementation](https://github.com/developit/preact/blob/master/src/vdom/functional-component.js#L12-L15), which accounts for ES2015 arrow functions with no prototype, and allows Components to not extend from Component:

```js
function isStatelessComponent(c) {
	return typeof c==='function' && !(c.prototype && c.prototype.render);
}
```

File: src/index.js [hunk 2]
```diff
@@ -129,11 +129,40 @@ for (let i=ELEMENTS.length; i--; ) {
...
+
+function statelessComponentHook(WrappedComponent) {
+	// Ensure the displayName has been set for debugging purposes
+	WrappedComponent.displayName = WrappedComponent.displayName
+		|| WrappedComponent.name
+		|| 'StatelessComponent';
+
+	function StatelessComponent(props) {
+		// Ensure default props have been applied to the props.
+		const p = WrappedComponent.defaultProps
```
Comment (developit) [inline src/index.js]: I believe defaultProps should already be applied by this point - stateless functions are invoked [here](https://github.com/developit/preact/blob/master/src/vdom/functional-component.js#L24), and their props are mutated prior to invocation [here](https://github.com/developit/preact/blob/master/src/vdom/index.js#L36-L45), which seems to apply `defaultProps`.

Thoughts?

**Pull request closed without merging**